### PR TITLE
feat(clickhouse): support generate TimestampTrunc, Variance, Stddev

### PR DESCRIPTION
**Pull request by longxiaofei**
References:

* https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions#date_trunc
* https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/varsamp
* https://clickhouse.com/docs/en/sql-reference/aggregate-functions/reference/stddevsamp

### CODE REVIEW

File: sqlglot/dialects/clickhouse.py [hunk 1]
```diff
@@ -761,6 +762,9 @@ class Generator(generator.Generator):
                 "SHA256" if e.text("length") == "256" else "SHA512", e.this
             ),
             exp.UnixToTime: _unix_to_time_sql,
+            exp.TimestampTrunc: timestamptrunc_sql,
+            exp.Variance: lambda self, e: self.func("varSamp", e.this),
```
Comment (tobymao) [inline sqlglot/dialects/clickhouse.py:766]: exp.Variance: rename_func("varSamp")
exp.Stddev: rename_func("stddevSamp")

Comment (longxiaofei) [reply sqlglot/dialects/clickhouse.py]: resolved

File: sqlglot/dialects/clickhouse.py [hunk 2]
```diff
@@ -158,6 +159,7 @@ class Parser(parser.Parser):
                 this=seq_get(args, 2), expression=seq_get(args, 1), unit=seq_get(args, 0)
             ),
             "DATE_FORMAT": _build_date_format,
+            "DATE_TRUNC": date_trunc_to_time,
```
Comment (georgesittas) [inline sqlglot/dialects/clickhouse.py:162]: Why do we need this?

**Review by georgesittas — COMMENTED** [review]
Thanks @longxiaofei. I've got one more question, LGTM otherwise.

File: sqlglot/dialects/clickhouse.py (hunk 2 above)
Comment (longxiaofei) [reply sqlglot/dialects/clickhouse.py:162]: Uh, I wanted to add a test case for generating date_trunc with the zone parameter.

I wanted to use CH -> CH to test it, and found that clickhouse cannot parse date_trunc.

Then I added this parser...

Comment (georgesittas) [reply sqlglot/dialects/clickhouse.py:162]: I just tried parsing it with CH and found no issue:

```python
>>> import sqlglot
>>> sqlglot.parse_one("DATE_TRUNC('HOUR', a, 'UTC')", "clickhouse")
DateTrunc(
  unit=Literal(this=HOUR, is_string=True),
  this=Column(
    this=Identifier(this=a, quoted=False)),
  zone=Literal(this=UTC, is_string=True))
>>> sqlglot.parse_one("DATE_TRUNC('HOUR', a, 'UTC')", "clickhouse").sql("clickhouse")
"DATE_TRUNC('HOUR', a, 'UTC')"
```

Comment (longxiaofei) [reply sqlglot/dialects/clickhouse.py]: I remembered it wrong, and I didn't check it before coding😭

File: tests/dialects/test_duckdb.py [hunk 3]
```diff
@@ -742,6 +742,28 @@ def test_duckdb(self):
...
+                "duckdb": "STDDEV(a)",
+                "clickhouse": "stddevSamp(a)",
+            },
+        )
+        self.validate_all(
+            "DATE_TRUNC('DAY', x)",
+            write={
+                "duckdb": "DATE_TRUNC('DAY', x)",
+                "clickhouse": "DATE_TRUNC('DAY', x)",
+            },
```
Comment (georgesittas) [inline tests/dialects/test_duckdb.py:764]: Didn't we have a test with a time zone?

Comment (longxiaofei) [reply tests/dialects/test_duckdb.py:764]: added test cases in `test_dialect.py`

**Review by georgesittas — APPROVED** [review]

**Pull request merged**